contents/docs/libraries/next-js.md

```diff
@@ -332,25 +335,29 @@ import { PostHog } from 'posthog-node'
 export default function PostHogClient() {
   const posthogClient = new PostHog(process.env.NEXT_PUBLIC_POSTHOG_KEY, {
     host: process.env.NEXT_PUBLIC_POSTHOG_HOST,
+    flushAt: 1,
+    flushInterval: 0
   })
   return posthogClient
 }
 ```
 
-With this component, we can both send events in the body of a component, and get data from PostHog (such as feature flag evaluations) using a server-side function (named `getData()` here).
+> **Note:** Because our server-side `posthog-node` initializations are short-lived, we set `flushAt` to `1` and `flushInterval` to `0`. This ensures that events are sent immediately and not batched. We also need to call `await posthog.shutdownAsync()` after sending events to flush them.
+
+With this `PostHogClient` component, we can both send events and fetch data from PostHog (such as feature flag evaluations) without client-side requests.
 
 ```js
 import Link from 'next/link'
 import PostHogClient from '../posthog'
 
 export default async function About() {
 
-  const flags = await getData();
-
-  posthog.capture({
-    distinctId: 'user_distinct_id', // replace with a user's distinct ID
-    event: 'server-side event'
-  })
+  const posthog = PostHogClient()
+  const flags = await posthog.getAllFlags(
+    'user_distinct_id' // replace with a user's distinct ID
+  );
+  await posthog.shutdownAsync()
+  return flags
```

Comment:
why `return flags` here? We wouldn't reach the return statement below then? Looks like a bug 

Also, Im not clued up on the latest NextJs changes, but is this a server side rendered component now? Thats why we're using posthog-node here, right?

Reply:
typo! yes, this is server-rendered.